src/System.Management.Automation/CoreCLR/CorePsPlatform.cs

```diff
@@ -204,6 +204,29 @@ internal static class CommonEnvVariableNames
 #endif
         }
 
+        /// <summary>
+        /// Get a temporary directory to use, needs to be unique to avoid collision
+        /// </summary>
+        public static string GetTemporaryDirectory()
```

Comment:
I open Issue to enhance `New-TemporaryFile` cmdlet to return a temporary directory.
So maybe we move the code to PSUtils.cs?